frontends/web/src/routes/settings/manage-accounts.tsx

```diff
@@ -43,93 +43,19 @@ interface ManageAccountsProps {
-          <div
-            className={`${style.acccountLink} ${active ? style.accountActive : ''}`}
-            onClick={() => active && route(`/account/${account.code}`)}>
-            <Logo stacked active={account.active} className={`${style.coinLogo} m-right-half`} coinCode={account.coinCode} alt={account.coinUnit} />
-            <span className={!account.active ? style.accountNameInactive : ''}>
-              {account.name}
-              {' '}
-              <span className="unit">({account.coinUnit})</span>
-            </span>
-          </div>
-          <div className="flex flex-items-center">
-            {!account.active ? <p className={`text-small ${style.disabledText}`}>{t('generic.enabled_false')}</p> : null}
-            <Button
-              className={style.editBtn}
-              onClick={() => this.setState({ currentlyEditedAccount: account })}
-              transparent>
-              <EditActive />
-              <span className="hide-on-small">
-                {t('manageAccounts.editAccount')}
-              </span>
-            </Button>
-          </div>
-          {active && account.coinCode === 'eth' ? (
-            <div className={style.tokenSection}>
-              <div className={`${style.tokenContainer} ${tokensVisible ? style.tokenContainerOpen : ''}`}>
-                {this.renderTokens(account.code, account.activeTokens)}
-              </div>
-              <Button
-                className={`${style.expandBtn} ${tokensVisible ? style.expandBtnOpen : ''}`}
-                onClick={() => this.toggleShowTokens(account.code)}
-                transparent>
-                {t(tokensVisible ? 'manageAccounts.settings.hideTokens' : 'manageAccounts.settings.showTokens', {
-                  activeTokenCount: `${account.activeTokens?.length || 0}`
-                })}
-              </Button>
-            </div>
-          ) : null}
-        </div>
-      );
-    });
-  };
-
-  private toggleAccount = (accountCode: accountAPI.AccountCode, active: boolean) => {
-    return backendAPI.setAccountActive(accountCode, active).then(({ success, errorMessage }) => {
-      if (!success && errorMessage) {
-        alertUser(errorMessage);
-      }
-    });
-  };
-
-  private toggleShowTokens = (accountCode: accountAPI.AccountCode) => {
-    this.setState(({ showTokens }) => ({
-      showTokens: {
-        ...showTokens,
-        [accountCode]: (accountCode in showTokens) ? !showTokens[accountCode] : true,
-      }
-    }));
-  };
+export const ManageAccounts = ({ accounts, devices, hasAccounts }: Props) => {
+  const { t } = useTranslation();
+  const [editErrorMessage, setEditErrorMessage] = useState<string | undefined>(undefined);
+  const [showTokens, setShowTokens] = useState<TShowTokens>({});
+  const [currentlyEditedAccount, setCurrentlyEditedAccount] = useState<accountAPI.IAccount | undefined>(undefined);
 
-  private erc20Tokens: accountAPI.Terc20Token[] = [
+  const erc20Tokens: accountAPI.Terc20Token[] = [
```

Comment:
Since you are here, you could define this before the component  and also make readonly

```suggestion
  const erc20Tokens: Readonly<accountAPI.Terc20Token[]> = [
```